2017/opmodes/DaisyBeaconAutonomous.java

```diff
@@ -0,0 +1,211 @@
+    private AutonomousAction actionChoice = AutonomousAction.LAUNCH_2;
+    private AutonomousBeacon beaconChoice = AutonomousBeacon.BEACON_1;
+
+    public enum Alliance {
+        RED,
+        BLUE,
+    }
+
+    public enum AutonomousPath {
+        CORNER_PARK,
+        CENTER_PARK,
+        STAY,
+    }
+
+    public enum AutonomousAction {
+        LAUNCH_1,
+        LAUNCH_2,
+    }
+
+    public enum AutonomousBeacon {
+        BEACON_1,
+        BEACON_2,
+    }
+
+    @Override
+    public void handleEvent(RobotEvent e) {
+        if (e instanceof GamepadTask.GamepadEvent) {
+            GamepadTask.GamepadEvent event = (GamepadTask.GamepadEvent) e;
+
+            if (event.kind == GamepadTask.EventKind.BUTTON_X_DOWN) {
+                selectAlliance(Alliance.BLUE);
+                ptt.addData("ALLIANCE", "Blue");
+            } else if (event.kind == GamepadTask.EventKind.BUTTON_B_DOWN) {
+                selectAlliance(Alliance.RED);
+                ptt.addData("ALLIANCE", "Red");
+            } else if (event.kind == GamepadTask.EventKind.LEFT_TRIGGER_DOWN) {
+                pathChoice = AutonomousPath.CORNER_PARK;
+                ptt.addData("AUTONOMOUS", "Corner Park");
+            } else if (event.kind == GamepadTask.EventKind.RIGHT_TRIGGER_DOWN) {
+                pathChoice = AutonomousPath.CENTER_PARK;
+                ptt.addData("AUTONOMOUS", "Center Park");
+            } else if (event.kind == GamepadTask.EventKind.LEFT_BUMPER_DOWN) {
+                actionChoice = AutonomousAction.LAUNCH_1;
+                ptt.addData("AUTONOMOUS", "Launch 1 Ball");
+            } else if (event.kind == GamepadTask.EventKind.RIGHT_BUMPER_DOWN) {
+                actionChoice = AutonomousAction.LAUNCH_2;
+                ptt.addData("AUTONOMOUS", "Launch 2 Balls");
+            } else if (event.kind == GamepadTask.EventKind.BUTTON_A_DOWN) {
+                beaconChoice = AutonomousBeacon.BEACON_1;
+                ptt.addData("BEACON", "Claim 1 Beacon");
+            } else if (event.kind == GamepadTask.EventKind.BUTTON_Y_DOWN) {
+                beaconChoice = AutonomousBeacon.BEACON_2;
+                ptt.addData("BEACON", "Claim 2 Beacons");
+            }
+        }
+
+        if (e instanceof SingleShotTimerTask.SingleShotTimerEvent) {
+            conveyor.setPower(0);
+            addTask(runToPositionTask);
+        } else if (e instanceof RunToEncoderValueTask.RunToEncoderValueEvent) {
+            RunToEncoderValueTask.RunToEncoderValueEvent event = (RunToEncoderValueTask.RunToEncoderValueEvent) e;
+            if (event.kind == RunToEncoderValueTask.EventKind.DONE) {
+                if (!launched && actionChoice == AutonomousAction.LAUNCH_2) {
+                    conveyor.setPower(0.5);
+                    addTask(stt);
+                    launched = true;
+                } else {
+                    addTask(deadReckonBeaconTask);
+                }
+            }
+        }
+    }
```

Comment:
Nice.  Remind me to show you how to use anonymous class definitions with task specific event handlers.  That would make this robot event handler a little cleaner.  In the meantime, you should consider refactoring this and having separate methods for the different types of events.  e.g.

```
if (e instance of GamepadTask.GamepadEvent) {
    handleGamepadEvent(e);
} else if (e instance of SingleShotTimerTask.SingleShotTimerEvent) {
    handleSingleShotTimerEvent(e);
} else if (e instance of RunToEncoderValueTask.RunToEncoderValueEvent) {
    handleEncoderValueEvent(e);
}
```

I really like your gamepad based selection mechanism.